bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -142,8 +143,13 @@ def _date_from_json(value, field):
 def _time_from_json(value, field):
     """Coerce 'value' to a datetime date, if set or not nullable"""
     if _not_null(value, field):
-        # value will be a string, in HH:MM:SS form.
-        return _time_from_iso8601_time_naive(value)
+        if len(value) == 8:  # HH:MM:SS
```

Comment:
Should these be regexes instead of length checks?

Reply:
I think not:  `datetime.datetime.strptime` will raise a `ValueError` if the value is bogus:  I'm just using the length to select which format string applies.